ui/components/ui/new-network-info/new-network-info.js

```diff
@@ -0,0 +1,231 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { I18nContext } from '../../../contexts/i18n';
+import Popover from '../popover';
+import Button from '../button';
+import Identicon from '../identicon/identicon.component';
+import { NETWORK_TYPE_RPC } from '../../../../shared/constants/network';
+import Box from '../box';
+import {
+  ALIGN_ITEMS,
+  COLORS,
+  DISPLAY,
+  FONT_WEIGHT,
+  TEXT_ALIGN,
+  TYPOGRAPHY,
+} from '../../../helpers/constants/design-system';
+import Typography from '../typography';
+import { TOKEN_API_METASWAP_CODEFI_URL } from '../../../../shared/constants/tokens';
+import fetchWithCache from '../../../helpers/utils/fetch-with-cache';
+import {
+  getNativeCurrencyImage,
+  getUseTokenDetection,
+} from '../../../selectors';
+import { IMPORT_TOKEN_ROUTE } from '../../../helpers/constants/routes';
+import Chip from '../chip/chip';
+
+const NewNetworkInfo = ({ closePopup = null }) => {
+  const t = useContext(I18nContext);
+  const history = useHistory();
+  const [tokenDetectionSupported, setTokenDetectionSupported] = useState(false);
+  const autoDetectToken = useSelector(getUseTokenDetection);
+  const primaryTokenImage = useSelector(getNativeCurrencyImage);
+
+  const currentProvider = useSelector((state) => ({
```

Comment:
We could use the `getProvider` selector here

Reply:
Fixed in: 11eaf89a7c325b3630324c49fef897cb2c4e6a42